src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1444,4 +1445,109 @@ public static void modifyAttributes()
             FORGE_ATTRIBUTES.put(k, newMutable.create());
         });
     }
+
+    private static final String ALL_DIMS_KEY = "all_dimensions_forge";
+    private static final String CODEC_DIM_KEY = "dimensions"; // key for the dim reg in DimensionGeneratorSettings
+
+    /**
+     * Codec that tries to decode the registry of all dimensions.
+     */
+    public static MapCodec<SimpleRegistry<Dimension>> tryAllDimensionsCodec(Codec<SimpleRegistry<Dimension>> base)
+    {
+        return new MapCodec<SimpleRegistry<Dimension>>()
+        {
+            @Override
+            public <T> DataResult<SimpleRegistry<Dimension>> decode(DynamicOps<T> ops, MapLike<T> input)
+            {
+                return base.parse(ops, input.get(ALL_DIMS_KEY) != null ? input.get(ALL_DIMS_KEY) : input.get(CODEC_DIM_KEY));
+            }
+
+            @Override
+            public <T> RecordBuilder<T> encode(SimpleRegistry<Dimension> input, DynamicOps<T> ops, RecordBuilder<T> prefix)
+            {
+                return prefix.add(CODEC_DIM_KEY, base.encodeStart(ops, input));
+            }
+
+            @Override
+            public <T> Stream<T> keys(DynamicOps<T> ops)
+            {
+                return Stream.of(ops.createString(ALL_DIMS_KEY), ops.createString(CODEC_DIM_KEY));
+            }
+        };
+    }
+
+    public static final MapCodec<CompoundNBT> ALL_DIMENSIONS_CODEC = new MapCodec<CompoundNBT>()
+    {
+        @Override
+        public <T> DataResult<CompoundNBT> decode(DynamicOps<T> ops, MapLike<T> input)
+        {
+            T allDims = input.get(ALL_DIMS_KEY);
+            if (allDims != null)
+            {
+                INBT allNbt = allDims instanceof INBT ? (INBT) allDims : ops.convertTo(NBTDynamicOps.INSTANCE, allDims);
+                if (allNbt instanceof CompoundNBT)
+                    return DataResult.success((CompoundNBT)allNbt);
+            }
+            else // will happen when "porting" worlds to this version.
+            {
+                T curr = input.get(CODEC_DIM_KEY);
+                if (curr == null)
+                    return DataResult.error("Could not find \"dimensions\" object while decoding world gen settings, impossible!", new CompoundNBT());
+
+                INBT nbt = curr instanceof INBT ? (INBT) curr : ops.convertTo(NBTDynamicOps.INSTANCE, curr);
+                if (nbt instanceof CompoundNBT)
+                    return DataResult.success((CompoundNBT) nbt);
+            }
+            return DataResult.error("All dimensions tag is not a compound.", new CompoundNBT());
```

Comment:
This sentence is a bit confusing, maybe "Tag for all dimensions is not a compound." or "Tag representing all dimensions is not a compound."